Feature/redo os image usage

arbulu89 opened the pull request:
Some new implementations to the mechanism used to select the OS image for the machines.

It brings:

- Use PAYG images by default. We receive some issues because the users use default values (BYOS) and they don't have (or don't use) the registration code. For newcomers (or the most standard user), PAYG are better.
- Create new os_image variable as default OS image. Right now, we need to set the OS image for all of the nodes (hana, iscsi, etc). As most of the users will use the same OS for all the machines, having this option makes things easier, as you only need to change 1 variable. Specific customization is still possible of course.
- Unify naming. Now, all the variables for all providers follow the same syntax (`*_os_image).
- Change how Azure images are selected. I think that is easier to only have unique variable, instead of 4, that can be obtained from CLI commands output
- Reformat the terraform.tfvars.example to be more informative (I think I will split this file in terraform.tfvars.example.basic and terraform.tfvars.example.advanced in the future)

FYI @peteratsuse

Simranpal commented on azure/modules/drbd_node/main.tf:
```diff
@@ -187,10 +192,10 @@ resource "azurerm_virtual_machine" "drbd" {
+    publisher = var.drbd_image_uri != "" ? "" : module.os_image.publisher
+    offer     = var.drbd_image_uri != "" ? "" : module.os_image.offer
+    sku       = var.drbd_image_uri != "" ? "" : module.os_image.sku
+    version   = var.drbd_image_uri != "" ? "" : module.os_image.version
```
Do these properties of `os_image` need to be defined as well in the terraform tfvars? If so should we provide an example there?

Simranpal replied:
I can see the properties are defined in os_image module, but should this be taken as input from user

arbulu89 replied:
No. I changed the concept. Before, we asked the user for these values individually, and we had 4 variables. Now, I changed to a unique variable (`os_image`) that is composed by all the values separated by `:`.
Basically, the `os_image` module takes `os_image = "SUSE:sles-sap-15-sp2:gen2:latest"` and converts to `publisher = "SUSE"`, `offer = "sles-sap-15-sp2"`, etc.
I did this to reduce the amount of changes required to update a sles version

Simranpal commented on azure/modules/drbd_node/variables.tf:
```diff
@@ -45,24 +45,9 @@ variable "drbd_image_uri" {
-  type    = string
-  default = "latest"
+variable "os_image" {
+  description = "sles4sap image used to create the this module machines. Composed by 'Publisher:Offer:Sku:Version' syntax. Example: SUSE:sles-sap-15-sp2:gen2:latest"
```
create the this /  create this

Simranpal replied:
same for other places defining variable "os_image"

arbulu89 replied:
Thanks, I will have a look

Simranpal commented on azure/modules/hana_node/main.tf:
```diff
@@ -202,6 +202,11 @@ resource "azurerm_image" "sles4sap" {
 # hana instances
 
+module "os_image" {
+  source   = "../../modules/os_image_reference"
```
I guess you can use "`../os_image_reference`"

Simranpal commented on gcp/terraform.tfvars.example:
```diff
@@ -101,79 +170,51 @@ sap_hana_deployment_bucket = "MyHanaBucket/sapdata/sap_inst_media/51053381"
 
-# Run provisioner execution in background
-#background = true
+# In order to enable SBD, an ISCSI server is needed as right now is the unique option
```
`is needed as right now it is the only option`

Simranpal reviewed (changes requested):
@arbulu89 I have doubt about the `os_image` if we should add the example usage  of setting sku, version etc. Other than that some typos and corrections in variable description and terraform.tfvars.example. I made the suggestions in `aws/terraform.tfvars.example` but it should apply to other modules

MalloZup reviewed (approved):
+1

arbulu89 commented on aws/modules/monitoring/variables.tf:
```diff
@@ -11,7 +11,7 @@ variable "monitoring_enabled" {
   description = "The instance type of monitoring node."
   type        = string
-  default     = "t2.micro"
+  default     = "t3.micro"
```
There is not any documentation/suggestion. I changed this just after trial and error.
Depending on the nodes (`iscsi`), if we use `micro/small` types, the `xvdd` disk is not mounted. So we need to use large.
For the `monitoring` it doesn't really matter, and I think the cost is the same

arbulu89 commented on aws/terraform.tfvars.example:
```diff
@@ -17,38 +17,110 @@ aws_region = "eu-central-1"
+#    "sle-ha-geo/12.4/x86_64" = "<<REG_CODE>>"
+#}
+
+# Default os_image and os_owner. These values are not used if the specific values are not set (e.g.: hana_os_image)
```
Yes! Thanks for having a deep look XD

arbulu89 commented on aws/terraform.tfvars.example:
```diff
@@ -17,38 +17,110 @@ aws_region = "eu-central-1"
+# Provisioning log level (error by default)
+#provisioning_log_level = "info"
+
+# Enable all some pre deployment steps (disabled by default)
```
Yes, this needs a correction. I just copy/pasted without rewriting

arbulu89 commented on aws/terraform.tfvars.example:
```diff
@@ -17,38 +17,110 @@ aws_region = "eu-central-1"
+# true or false (default)
+#qa_mode = false
+
+# Execute HANA Hardware Configuration Check Tool to bench filesystems
```
No idea. This was set by the QA team. In any case, most probably we will remove the `qa_mode` in the future, so I wouldn't be so worried

arbulu89 commented:
@Simranpal Thank you so much for your nice review! I have addressed all the comments and applied your suggestions.
Have a new review when you have some time.

Simranpal reviewed (approved):
Thanks for the changes @arbulu89 looks good!
